### PR TITLE
New memory allocator feature, to measure the size of an allocated memory area. 

### PR DESCRIPTION
**Pull request by mnurczynski**
Added a kmsz function to the allocator, to check the size of an allocated memory area.
Function is properly documented in the kmalloc.h header file.

### CODE REVIEW

**Review by MateuszTk — APPROVED** [review]
👍

File: src/kernel/kmalloc.h [hunk 1]
```diff
@@ -62,3 +62,16 @@ extern void kmres(void* pointer, uint32_t size);
  * @return Pointer to the enlarged memory area. 0 if enlargment was not possible.
  */
 extern void* krealloc(void* pointer, uint32_t new_size);
+
+
+
+/**
+ * @brief Measures the size of an allocated area. If used on an unallocated memory space - the behaviour is undefined.
+ *
+ * @note Allocator allocates memory based on the buddy system, using blocks. Therefore actual size of an allocated area (which this function will return) will be larger than the size given in malloc. It is NOT a bug, it's a feature.
```
Comment (magistermaks) [inline src/kernel/kmalloc.h]: larger or equal than

File: src/kernel/entry.c [hunk 2]
```diff
@@ -20,8 +21,13 @@ void start() {
...
 
-//	kprintf("\e[2J%% Hello \e[1;33m%s\e[m wo%cld, party like it's \e[1m%#0.8x\e[m again!\n", "sweet", 'r', -1920);
-
+    kprintf("\e[2J%% Hello \e[1;33m%s\e[m wo%cld, party like it's \e[1m%#0.8x\e[m again!\n", "sweet", 'r', -1920);
+    void* ptr = kmalloc(50000000);
+    kprintf("\n%d", ptr);
+    void* ptr2 = kmalloc(1000);
+    kprintf("\n%d", ptr2);
+    kprintf("\n%d", kmsz(ptr));
+    halt();
```
Comment (magistermaks) [inline src/kernel/entry.c]: don't `halt()` in the middle of kernel init please

**Review by magistermaks — CHANGES_REQUESTED** [review]

File: src/kernel/kmalloc.h (hunk 1 above)
Comment (mnurczynski) [reply src/kernel/kmalloc.h]: Replaced ,,will be larger" with ,,could be larger".

File: src/kernel/entry.c (hunk 2 above)
Comment (mnurczynski) [reply src/kernel/entry.c]: I deleted the entire code that is meant for testing the new feature.

**Review by magistermaks — APPROVED** [review]

**Pull request merged**